core/src/main/java/com/linecorp/armeria/client/circuitbreaker/CircuitBreakerListener.java

```diff
@@ -21,6 +21,14 @@
  */
 public interface CircuitBreakerListener {
 
+    /**
+     * Invoked when the circuit breaker is initialized.
+     * The initial state is always {@link CircuitState.CLOSED}.
+     */
+    default void onInitialized(String circuitBreakerName) throws Exception {
+        onStateChanged(circuitBreakerName, CircuitState.CLOSED);
+    }
```

Comment:
How about adding `CircuitState initialState` as the second parameter, so that we can have a different type of circuit, which starts at HALF_OPEN or OPEN?

Reply:
Or please let me know if it doesn't make sense, of course. :smile:

Reply:
Yes you right, `initialState` is more general for other type of circuit.
Thank you. 😁